exporter/opentelemetry-exporter-otlp/src/opentelemetry/exporter/otlp/exporter.py

```diff
@@ -64,8 +73,12 @@ def _translate_key_values(key, value):
 
 
 def _get_resource_data(
-    sdk_resource_instrumentation_library_data, resource_class, name
-):
+    sdk_resource_instrumentation_library_data: Dict[
+        SDKResource, ResourceDataT
+    ],
+    resource_class: Callable[..., TypingResourceT],
```

Comment:
```suggestion
    resource_class: Type[TypingResourceT],
```

Reply:
To do this, we would need to constrain the type, because otherwise it typechecks the call against the object constructor and errors. Constraining the type however leads to more mypy errors because we instantiate an object of type TypingResourceT with a dynamically keyed dict and it can't correctly infer the argument types without constraining the name parameter and adding a few variable annotations to the kwarg dict. Which direction sounds more preferable?

Reply:
I see what you mean. I think a clean solution would be moving the dynamic part into the individual exporters (to make them static), but out of scope of this PR.